src/models/model.h

```diff
@@ -116,6 +121,7 @@ struct Model : std::enable_shared_from_this<Model> {
 
   std::unique_ptr<Config> config_;
   std::unique_ptr<OrtSessionOptions> session_options_;
+  std::unique_ptr<OrtSessionOptions> vision_session_options_;
```

Comment:
We should refactor this so that Model does not know about the session options needed by the derived model's state.

Maybe factor it out into a session options pool that can be queried to get the session options?

We can do that in a future PR.